src/plugins/issue-pr-link/index.js

```diff
@@ -0,0 +1,166 @@
+        );
+        return issue.state === "open";
+    } catch {
+        // Issue doesn't exist or we don't have access
+        return false;
+    }
+}
+
+/**
+ * Check if we already commented on this issue for this PR
+ * @param {ProbotContext} context Probot context object
+ * @param {number} issueNumber Issue number
+ * @param {number} prNumber PR number
+ * @returns {Promise<boolean>} True if we already commented
+ * @private
+ */
+async function hasExistingComment(context, issueNumber, prNumber) {
+    try {
+        const { data: comments } = await context.octokit.issues.listComments(
+            context.repo({ issue_number: issueNumber })
+        );
+
+        const botComments = comments.filter(comment => 
+            comment.user.type === "Bot" && 
+            comment.body.includes("[//]: # (issue-pr-link)") &&
+            comment.body.includes(`/pull/${prNumber}`)
+        );
+
+        return botComments.length > 0;
+    } catch {
+        // If we can't check comments, assume we haven't commented
+        return false;
+    }
+}
+
+/**
+ * Comment on issues referenced in the PR body
+ * @param {ProbotContext} context Probot context object
+ * @returns {Promise<void>}
+ * @private
+ */
+async function commentOnReferencedIssues(context) {
+    const { payload } = context;
+    const pr = payload.pull_request;
+    
+    if (!pr || !pr.body) {
+        return;
+    }
+
+    const issueNumbers = extractIssueNumbers(pr.body);
+    
+    if (issueNumbers.length === 0) {
+        return;
+    }
+
+    const prUrl = pr.html_url;
+    const prAuthor = pr.user.login;
+    const prNumber = pr.number;
+
+    // Comment on each referenced issue
+    for (const issueNumber of issueNumbers) {
+        try {
+            // Check if issue exists and is open
+            if (!(await isIssueOpenAndExists(context, issueNumber))) {
+                continue;
+            }
+
+            // Check if we already commented on this issue for this PR
+            if (await hasExistingComment(context, issueNumber, prNumber)) {
+                continue;
+            }
+
+            // Create the comment
+            await context.octokit.issues.createComment(
+                context.repo({
+                    issue_number: issueNumber,
+                    body: createCommentMessage(prUrl, prAuthor)
+                })
+            );
+        } catch (error) {
+            // Log error but continue with other issues
+            // eslint-disable-next-line no-console -- Logging errors is intentional
+            console.error(`Failed to comment on issue #${issueNumber}:`, error);
+        }
+    }
+}
+
+//-----------------------------------------------------------------------------
+// Robot
+//-----------------------------------------------------------------------------
+
+module.exports = robot => {
+    robot.on("pull_request.opened", commentOnReferencedIssues);
+    robot.on("pull_request.edited", commentOnReferencedIssues);
+};
```

Comment:
```suggestion
};

```

I think an EOL is needed here.